src/common/internet.rs

```diff
@@ -0,0 +1,310 @@
+}
+
+impl TryFrom<String> for InetAddr {
+    type Error = ();
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        InetAddr::try_from(value.as_str())
+    }
+}
+
+impl TryFrom<&str> for InetAddr {
+    type Error = ();
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        unimplemented!()
+    }
+}
+
+impl TryFrom<Vec<u8>> for InetAddr {
+    type Error = ();
+    fn try_from(value: Vec<u8>) -> Result<Self, Self::Error> {
+        InetAddr::try_from(&value[..])
+    }
+}
+
+impl TryFrom<&[u8]> for InetAddr {
+    type Error = ();
+    fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
+        match value.len() {
+            4 => {
+                let mut buf = [0u8; 4];
+                buf.clone_from_slice(value);
+                Ok(InetAddr::from(buf))
+            },
+            16 => {
+                let mut buf = [0u8; 16];
+                buf.clone_from_slice(value);
+                Ok(InetAddr::from(buf))
+            },
+            #[cfg(feature="use-tor")]
+            32 => {
+                let mut buf = [0u8; 32];
+                buf.clone_from_slice(value);
+                InetAddr::try_from(buf)
+            }
+            _ => Err(())
+        }
+    }
+}
+
+impl From<[u8; 4]> for InetAddr {
+    fn from(value: [u8; 4]) -> Self {
+        InetAddr::from(Ipv4Addr::from(value))
+    }
+}
+
+impl From<[u8; 16]> for InetAddr {
+    fn from(value: [u8; 16]) -> Self {
+        InetAddr::from(Ipv6Addr::from(value))
+    }
+}
+
+impl From<[u16; 8]> for InetAddr {
+    fn from(value: [u16; 8]) -> Self {
+        InetAddr::from(Ipv6Addr::from(value))
+    }
+}
+
+#[cfg(feature="use-tor")]
+impl TryFrom<[u8; TORV3_PUBLIC_KEY_LENGTH]> for InetAddr {
+    type Error = ();
+    fn try_from(value: [u8; 32]) -> Result<Self, Self::Error> {
+        Self::from_32bit_encoding(value).ok_or(())
+    }
+}
+
+/// Transport protocols that may be part of `TransportAddr`
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum Transport {
+    /// Normal TCP
+    TCP,
+
+    /// Normal UDP
+    UDP,
+
+    /// Multipath TCP version
+    MTCP,
+
+    /// More efficient UDP version under developent by Google and consortium of
+    /// other internet companies
+    QUIC,
+
+    // There are other rarely used protocols. Do not see any reason to add
+    // them to the LNP/BP stack for now, but it may appear in the future,
+    // so keeping them for referencing purposes:
+    /*
+    UDPLite,
+    SCTP,
+    DCCP,
+    RUDP,
+    */
+}
+
+impl TryFrom<String> for Transport {
+    type Error = ();
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        let mut val = value.to_lowercase();
+        val.push(':');
+        Ok(match &val[..4] {
+            "tcp:" => Transport::TCP,
+            "udp:" => Transport::UDP,
+            "mtcp" => Transport::MTCP,
+            "quic" => Transport::QUIC,
+            _ => Err(())?
+        })
```

Comment:
Not a huge fan of pushing and checking on the longer string. I would just use `starts_with()`